src/lib/utils.ts

```diff
@@ -4,14 +4,26 @@ import { EVENT_MAP } from './constants';
 
 const defaultHanlder = () => null;
 
-export const buildQueryString = (
-  config: Record<string, string | Record<string, string>>,
-): string => Object.keys(config).map((key) => {
-  if (typeof config[key] === 'string') {
-    return `${key}=${config[key]}`
-  }
-  return `${key}=${JSON.stringify(config[key])}`
-}).join('&');
+interface QueryObject {
+  [key: string]: string | number | boolean | QueryObject;
+}
+
+export const buildQueryString = (config: QueryObject, configKey?: string): string => {
+  const pairs: string[] = [];
+
+  Object.entries(config).forEach(([key, value]) => {
+    const nestedKey = configKey ? `${configKey}[${encodeURIComponent(key)}]` : encodeURIComponent(key);
+    const valueIsObject = typeof value === 'object' && value !== null && !Array.isArray(value);
```

Comment:
We could handle arrays right now so this won't be a problem in the future. I'm not sure how we represent arrays on query strings, but again, our dashboard does that, so it might be worth taking a look at that code